src/backend/distributed/planner/merge_planner.c

```diff
@@ -1207,12 +1255,19 @@ ErrorIfMergeQueryQualAndTargetListNotSupported(Oid targetRelationId, Query *orig
 							   "supported in MERGE sql with distributed tables")));
 	}
 
+#if PG_VERSION_NUM >= PG_VERSION_17
+	Node *joinCondition = (Node *) originalQuery->mergeJoinCondition;
+#else
+	Node *joinCondition = (Node *) originalQuery->jointree->quals;
+#endif
```

Comment:
Just noting that there's a similar _#if - #else - #endif_ on lines 752 - 756 so it could be refactored into a function ?

Reply:
Done.